src/toil/batchSystems/lsf.py

```diff
@@ -21,262 +21,133 @@
+            logger.debug("Cant determine exit code for job or job still running: " + str(job))
+            return None
 
-    def getRunningBatchJobIDs(self):
-        """Gets a map of jobs (as jobIDs) currently running (not just waiting) 
-        and a how long they have been running for (in seconds).
-        """
-        times = {}
-        currentjobs = set()
-        for x in self.getIssuedBatchJobIDs():
-            if x in self.lsfJobIDs:
-                currentjobs.add(self.lsfJobIDs[x])
+        def prepareBsub(self, cpu, mem, jobID):
+            """
+            Make a bsub commandline to execute.
+
+            params:
+              cpu: number of cores needed
+              mem: number of bytes of memory needed
+            """
+            if mem:
+                if per_core_reservation():
+                    mem = parse_memory(float(mem)/1024**3/int(cpu))
+                else:
+                    mem = parse_memory(float(mem)/1024**3)
+                bsubMem = '-R "select[type==X86_64 && mem > ' + str(mem) + '] '\
+                    'rusage[mem=' + str(mem) + ']" -M' + str(mem)
             else:
-                #not yet started
-                pass
-        process = subprocess.Popen(["bjobs"], stdout = subprocess.PIPE)
-
-        for curline in process.stdout:
-            items = curline.strip().split()
-            if (len(items) > 9 and (items[0]) in currentjobs) and items[2] == 'RUN':
-                jobstart = "/".join(items[7:9]) + '/' + str(date.today().year)
-                jobstart = jobstart + ' ' + items[9]
-                jobstart = time.mktime(time.strptime(jobstart,"%b/%d/%Y %H:%M"))
-                jobstart = time.mktime(time.strptime(jobstart,"%m/%d/%Y %H:%M:%S"))
-                times[self.jobIDs[(items[0])]] = time.time() - jobstart
-        return times
+                bsubMem = ''
+            cpuStr = '' if cpu is None else '-n ' + str(int(cpu))
+            bsubline = ["bsub", bsubMem, cpuStr,"-cwd", ".", "-o", "/dev/null",
+                    "-e", "/dev/null", '-J', 'toil_job_' + str(jobID)]
+            lsfArgs = os.getenv('TOIL_LSF_ARGS')
+            if lsfArgs:
+                bsubline.extend(lsfArgs.split())
+            return bsubline
 
-    def getUpdatedBatchJob(self, maxWait):
-        try:
-            sgeJobID, retcode = self.updatedJobsQueue.get(timeout=maxWait)
-            self.updatedJobsQueue.task_done()
-            jobID, retcode = (self.jobIDs[sgeJobID], retcode)
-            self.currentjobs -= {self.jobIDs[sgeJobID]}
-        except Empty:
-            pass
-        else:
-            return jobID, retcode, None
-
-    def getWaitDuration(self):
-        """We give parasol a second to catch its breath (in seconds)
+    @classmethod
+    def getWaitDuration(cls):
+        """We give LSF a second to catch its breath (in seconds)
         """
-        #return 0.0
         return 15
 
-    @classmethod
-    def getRescueBatchJobFrequency(cls):
```

Comment:
Just to note - This function doesn't seem to be used and could probably be removed everywhere. There is a test for it, but LSF isn't run currently for our CI tests.

Reply:
Nevermind, I see that it is in abstractGridEngineBatchSystem.